### DISCUSSION TITLE
Request for spell checker extension

### DISCUSSION THREAD

**Discussion by aliahmed102** [zed-industries/zed#15662] [Feature Requests]
I want a feature for check English typo! I used to use an extension in vs code called spell checker it was so useful. when a word spelling is incorrect it suggest then by clicking **ctrl + .** it shows the solutions.

**Comment by nathany-copia** [comment]
I've been using Code Spell Checker with Visual Studio Code, which is great because it understands camelCase, etc.
https://github.com/streetsidesoftware/vscode-spell-checker

It uses a library called cSpell, which is mostly TypeScript. Maybe it would be possible to turn that into a WebAssembly extension for Zed. I'm not sure.

Alternatively, Sublime Text uses a C++ library called Hunspell. But the existing Rust bindings don't appear to be well maintained.
https://www.sublimetext.com/docs/spell_checking.html

It would be nice to have at least basic spell checking for writing Markdown files, otherwise I'd likely use a different editor for that task. Spell checking that understands code would be amazing!

**Comment by bajrangCoder** [comment]
For zed this LSP protocol based spell checker can be used using extension:

https://github.com/valentjn/ltex-ls

**Reply by arscynic** [reply]
This works excellently, thank you.

**Comment by notpeter** [comment]
You might be interested in [BaptisteRoseau/zed-typos](https://github.com/BaptisteRoseau/zed-typos) which bundles [tekumara/typos-lsp](https://github.com/tekumara/typos-lsp).  It's focus is detecting typos with (nearly) no false positives, it's not a spell-checker but it definitely works. We use typos in CI too, so it's nice to catch things before committing.

**Comment by gianpaj** [comment]
we have this now https://github.com/mantou132/zed-cspell

This can be mark as answered

**Comment by vovavili** [comment]
[Codebook](https://zed.dev/extensions/codebook) and [Harper](https://github.com/zed-extensions/harper) are pretty decent. Harper is basically a Rust-based local-only alternative for Grammarly and LanguageTool.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
